[PyTorch] Devirtualize TensorImpl::sizes() with macro

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50176 [PyTorch] Devirtualize TensorImpl::sizes() with macro**

UndefinedTensorImpl was the only type that overrode this, and IIUC we don't need to do it.

Differential Revision: [D25817370](https://our.internmc.facebook.com/intern/diff/D25817370/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25817370/)!

swolchok committed:
[PyTorch] Devirtualize TensorImpl::sizes() with macro

UndefinedTensorImpl was the only type that overrode this, and IIUC we don't need to do it.

Differential Revision: [D25817370](https://our.internmc.facebook.com/intern/diff/D25817370/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25817370/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Devirtualize TensorImpl::sizes() with macro"


UndefinedTensorImpl was the only type that overrode this, and IIUC we don't need to do it.

Differential Revision: [D25817370](https://our.internmc.facebook.com/intern/diff/D25817370/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25817370/)!

[ghstack-poisoned]

ezyang reviewed (approved):
Guessing that `sizes_` is still initialized in undefined tensor (so we don't return garbage)

swolchok committed:
Update on "[PyTorch] Devirtualize TensorImpl::sizes() with macro"


UndefinedTensorImpl was the only type that overrode this, and IIUC we don't need to do it.

Differential Revision: [D25817370](https://our.internmc.facebook.com/intern/diff/D25817370/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25817370/)!

[ghstack-poisoned]
